hooks/src/index.js

```diff
@@ -348,6 +351,21 @@ export function useErrorBoundary(cb) {
 	];
 }
 
+export function useId() {
+	const state = getHookState(currentIndex++, 11);
+	if (!state._id) {
+		currentIdCounter++;
+
+		state._id =
+			'_P' +
+			currentInternal._rootId +
+			(currentInternal._parent._children.indexOf(currentInternal) +
+				currentInternal._depth +
```

Comment:
_domDepth might be more reliable here as sometimes people wrap their ssr with a wrapper for the routing, ...